extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml

```diff
@@ -0,0 +1,123 @@
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: azure-policy-patch-label-cluster-role
+  annotations:
+    "helm.sh/hook-weight": "25"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+rules:
+- apiGroups: [""]
+  resources: ["namespaces"]
+  verbs: ["get", "patch"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: azure-policy-patch-label-role-binding
+  namespace: kube-system
+  annotations:
+    "helm.sh/hook-weight": "30"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+subjects:
+- kind: ServiceAccount
+  name: azure-policy-patch-label-account
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: azure-policy-patch-label-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: azure-policy-patch-label-role-binding
+  namespace: azure-arc
+  annotations:
+    "helm.sh/hook-weight": "30"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+subjects:
+- kind: ServiceAccount
+  name: azure-policy-patch-label-account
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: azure-policy-patch-label-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: azure-policy-patch-label
+  namespace: kube-system
+  annotations:
+    "helm.sh/hook-weight": "35"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  template:
+    metadata:
+      labels:
+        app: azure-policy-patch-label
+    spec:
+      serviceAccountName: azure-policy-patch-label-account
+      automountServiceAccountToken: true
+      restartPolicy: Never
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: azure-policy-patch-label
+          image: {{ .Values.alpine.image.name }}:{{ .Values.alpine.image.tag }}
+          imagePullPolicy: IfNotPresent
+          command: 
+            - "/bin/sh"
+            - "-ec"
+            - |
+              set -o pipefail
+              token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+              namespaces="kube-system azure-arc"
+              for namespace in ${namespaces}
+              do
+                echo "Retrieving namespace ${namespace} details"
+                namespaceDetails=$(curl -s -X GET -k https://kubernetes.default.svc/api/v1/namespaces/${namespace} \
+                -H "Authorization: Bearer ${token}" \
+                -H "Content-Type: application/json" \
+                -H "Accept: application/json")
+                kind=$(echo $namespaceDetails | jq .kind)
+                if [ "$kind" = "\"Namespace\"" ]
```

Comment:
I assume this check is to ensure an error resource isn't returned. Is this correct?

Reply:
Yes, that's right.